### ISSUE TITLE
Allow users to edit and disable right click context menu contributions from extensions

### ISSUE THREAD

**Issue by phgmacedo** [microsoft/vscode#120395]
Currently, any installed extension can add entries to vscode context menu as they please. 

For exemple, the python extension has a dependency on the jupyter extension, and they both add their own context menu items, even if users don't want it. Here is what I get in my python file context menu: 

![113361745-01e73480-9323-11eb-9a6c-448ba420f9b0](https://user-images.githubusercontent.com/2473545/113369357-8fcd1a80-9337-11eb-978f-95a1bf47496d.png)

It seems they do very similar things, and users are forced to have one even if they don't use the other. 

Usually, vscode developers would treat this as an extension specific problem, and ask the user to open an issue at the extension repo, but in this case I tried opening an issue both at the [python repo](https://github.com/microsoft/vscode-python/issues/15834), and at the [jupyter extension repo](https://github.com/microsoft/vscode-jupyter/issues/5387). 

Both issues were closed without any discussion, and they are from very popular vscode extensions, making it very difficult for users to disable these menu contributions without monkey patching the extension package.json file, or maintaining a fork of the extensions. 

It would be great for vscode to provide a way for users to disable these unwanted contributions.

**Comment by phgmacedo** [comment]
I'll also add that in the case for these extensions, even microsoft's [own team have reported that multiple people have complained about the issue ](https://github.com/microsoft/vscode-jupyter/issues/5387#issuecomment-812266034), and also said the problem will not be addressed by the extensions, so it looks to me like a vscode problem.

[Context menu issues](https://github.com/microsoft/vscode/issues?q=is%3Aissue+context+menu+is%3Aopen) seem to be very popular, and I could not find any specific issue asking for context menu contribution disablement.

**Comment by phgmacedo** [comment]
The closed thing I could find was an issue where @generic-user [complains about the ridiculous menu bloat](https://github.com/microsoft/vscode/issues/40239#issuecomment-808255140) they have in the extensions disable issue.

**Comment by IllusionMH** [comment]
Looks like duplicate of #9285

**Comment by phgmacedo** [comment]
The replies I got from @claudiaregio from microsoft's own team are extremely tone deaf. [Their final position is](https://github.com/microsoft/vscode-jupyter/issues/5387#issuecomment-812274494): 

> users do not have to use or click on any options/features made available to them if they do not wish to use them

Unfortunately, users still have to see the menu items. :(

**Comment by phgmacedo** [comment]
> Looks like duplicate of #9285 

It's not a duplicate because the other issue is more like a request to allow extension/users to add toolbar. 

In this case, it's actually the opposite: I want to remove the bloat that the jupyter extension has added to my context menu.

**Comment by gjsjohnmurray** [comment]
@phgmacedo with respect, the referenced issue is titled "User configurable menus" and was opened by a key member of the team. It is a clear duplicate, and forking the discussion by keeping this one open isn't going to help IMO.

/duplicate

**Comment by phgmacedo** [comment]
> @phgmacedo with respect, the referenced issue is titled "User configurable menus" and was opened by a key member of the team. It is a clear duplicate, and forking the discussion by keeping this one open isn't going to help IMO.
> 
> /duplicate 

I searched for the issue before I opened mine, and I did not find it because it was very old and there are a over 50 pages of issues with context menu in their title. 

Regardless, I believe that my specific issue has a smaller scope, and could be easier to implement. 

The other one is quite general and talks about all sorts of menus. 

Anyways, I'm fine with duping it, the important thing would be to actually fix the problem I described.
